WcaOnRails/lib/advance_to_next_round_requirement.rb

```diff
@@ -1,22 +1,35 @@
 # frozen_string_literal: true
 
 class AdvanceToNextRoundRequirement
+  include ActiveModel::Validations
```

Comment:
This class tries to be three different things at once.

It would probably help a lot to make it a base class to three simple classes.